src/EventSubscriber/ApqSubscriber.php

```diff
@@ -49,6 +49,10 @@ public function onBeforeOperation(OperationEvent $event): void {
       if ($queryHash !== $computedQueryHash) {
         throw new Error('Provided sha does not match query');
       }
+      // Add cache context for dynamic page cache compatibility.
+      $event->getContext()->addCacheContexts(
+        ['url.query_args:variables', 'url.query_args:extensions']
+      );
```

Comment:
The problem is not specific to APQ. We are using a custom persisted query plugin, and it suffers from the same problem. I think this should go to a more generic place. Every GraphQL query that is run via `GET` will face this issue.

Reply:
Actually, I would just rename this event subscriber and loosen the condition at the top.